src/client.rs

```diff
@@ -1468,35 +1345,29 @@ mod test {
             ////
             // Now bump the root to version 3
 
-            client
-                .remote
-                .store_metadata(
-                    &root_path,
-                    &MetadataVersion::Number(2),
-                    &root2.to_raw().unwrap(),
-                )
-                .await
-                .unwrap();
-
-            client
-                .remote
-                .store_metadata(&root_path, &MetadataVersion::None, &root2.to_raw().unwrap())
-                .await
-                .unwrap();
-
-            client
-                .remote
-                .store_metadata(
-                    &root_path,
-                    &MetadataVersion::Number(3),
-                    &root3.to_raw().unwrap(),
-                )
+            // Make sure the version 2 is signed by version 1's keys.
+            let repo_keys2 = RepoKeys {
+                root_keys: vec![&KEYS[0], &KEYS[1]],
+                snapshot_keys: vec![&KEYS[0], &KEYS[1], &KEYS[2]],
+                targets_keys: vec![&KEYS[0], &KEYS[1], &KEYS[2]],
+                timestamp_keys: vec![&KEYS[0], &KEYS[1], &KEYS[2]],
+            };
+            let _root2 = RepoBuilder::new(repo_keys2, &repo)
+                .with_root(|b| b.version(2).expires(Utc.ymd(2038, 1, 1).and_hms(0, 0, 0)))
+                .commit()
```

Comment:
the test previously kept the same targets/snapshot/timestamp metadata and just regenerated the root metadata, but those are being re-built now.  I'm not sure if signatures are guaranteed to be the same if all the inputs are the same.  Timestamp would still indicate that snapshot version 1 is the most recent, so the client wouldn't notice clobbered remote metadata.

Reply:
That's a good point. I believe it depends on the signature algorithm. I think ed25519 signatures are consistent, but rsa signatures are (or can be) non-deterministic.

However, it looks like it might be a little complicated to clean things up to support that. Would you be okay if we address this in a future PR?